### PR TITLE
[filebeat] introduce dnsConfig values for the containers

### PR DESCRIPTION
**Pull request by orong-pp**
Signed-off-by: Oron Gola <oron.gola@perception-point.io>

- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [x] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Comment by orong-pp** [comment]
cla/check

**Comment by orong-pp** [comment]
hi
@jmlrt @Crazybus  please review

**Comment by orong-pp** [comment]
### some details:
**the issue:**
when kubernetes runs a dns query in one of its pods, it first tries to look for the record in its internal namespaces. it can take 4-5 queries that search different options before we actually get to the final answer
in case we query an external website like spot.io we get 4-5 times more queries than we actually need and this puts unwanted load on coredns pods.

the details for this are all explained here
https://dev.to/mrkaran/dns-lookups-in-kubernetes-5cm1

**solution:**
so to deal with this issue one can define the ndots dns configuration option
the default is 5 and it means that only urls with 5 or more dots (.) in it will be queried directly from external dns and not go through 4 redundant dns queries

to set the ndots config option lower I added this pull request and made it possible to control this value and set it lower, so that all queries are direct and no internal search is made

**dnsConfig documentation:**
this is how dnsConfig should be configured according to kubernetes docs
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/

**Comment by orong-pp** [comment]
I have now signed the CLA

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

File: filebeat/values.yaml [hunk 1]
```diff
@@ -51,6 +51,7 @@ envFrom: []
 # Root directory where Filebeat will write data to in order to persist registry data across pod restarts (file position and other metadata).
 hostPathRoot: /var/lib
 hostNetworking: false
+dnsConfig: {}
```
Comment (jmlrt) [inline filebeat/values.yaml:54]: adding an example here would be great
```suggestion
dnsConfig: {}
# options:
#   - name: ndots
#     value: "2"
```

**Review by jmlrt — APPROVED** [review]
LGTM⛴, just a suggestion about adding a commented example in `values.yaml`.

**Comment by orong-pp** [comment]
jenkins test this please

**Comment by orong-pp** [comment]
@jmlrt @Crazybus please review

**Review by jmlrt — APPROVED** [review]
LGTM⛴

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
backported to `6.8`, `7.9` and `7.x` branches

**Pull request merged**